signing_clients/apps.py

```diff
@@ -367,19 +385,43 @@ def make_signed(self, signature, outpath=None, sigpath=None):
                     zout.writestr('META-INF/ids.json', self.ids)
 
 
-# This is basically a dumbed down version of M2Crypto.SMIME.load_pkcs7 but
-# that reads DER instead of only PEM formatted files
+class SignatureInfo(object):
+
+    def __init__(self, pkcs7):
+        if isinstance(pkcs7, SignatureInfo):
+            # Allow passing around SignatureInfo objects to avoid
+            # re-reading the signature every time.
+            self.content = pkcs7.content
+        else:
+            self.content = cms.ContentInfo.load(pkcs7).native['content']
+
+    @property
+    def issuer_serial_number(self):
+        return self.signer_info['sid']['serial_number']
```

Comment:
nit on naming: `issuer` may be confusing, as what you're returning is the serial number of the end entity, not the intermediate CA that issued the end entity. `signer_serial_number` may be more accurate.